elixir/apps/api/lib/api/sockets.ex

```diff
@@ -116,6 +116,12 @@ defmodule API.Sockets do
     {location_region, location_city, {location_lat, location_lon}} =
       API.Sockets.load_balancer_ip_location(x_headers)
 
+    user_agent =
+      case Base.decode64(user_agent) do
+        {:ok, ua} -> ua
+        :error -> user_agent # for backwards compatibility
+      end
```

Comment:
I know it's _probably_ not a problem but automatically decoding or not decoding feels weird. Like I don't see a bug but it's the type of place that it seems like a bug could be in.

If we're outside the standard and controlling it anyway, could we make it a new header like `X-Base64-User-Agent` or `Firezone-User-Agent` and check that one first?

Reply:
Actually thinking about this a bit more, I think @AndrewDryga's suggestion to normalize might be better.

Random headers can be dropped from secure web gateways and other proxies.

The header we currently send isn't really valid either:

```
Mac OS/Version 14.5 (Build 23F79) (arm64;23.5.0;) connlib/1.1.1
```

should be

```
Mac OS/Version 14.5 (Build 23F79) connlib/1.1.1 (arm64;23.5.0;)
```

The syntax is `product/version (notes) thing/version (notes)` repeated it looks like:

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent

Reply:
It's just a little of work to fix this for Android and macOS/iOS. I'll take a peek at how we're building the user agent for others too.